### PR TITLE
Add binary, image, and main properties to WhiskActionMetaData

### PR DESCRIPTION
**Pull request by dubeejw**
This PR adds a binary, image, and main properties to a WhiskActionMetaData.

Values that should be shown in WhiskActionMetaData:
- string exec
	- kind, binary, main

- attach exec
	- kind, binary, main

- blackbox exec
	- kind, image, binary, main

- sequence exec
	- kind, components

### CODE REVIEW

**Comment by dubeejw** [comment]
PG3 1544 ⏳

**Comment by dubeejw** [comment]
Values that should be shown in WhiskActionMetaData:
- string exec
	-kind, binary, main

- attach exec
	- kind, binary, main

- blackbox exec
	- kind, image, binary, main

**Comment by dubeejw** [comment]
PG3 1548 🔵

File: tests/src/test/scala/whisk/core/controller/test/WebActionsApiTests.scala [hunk 1]
```diff
@@ -190,7 +190,7 @@ trait WebActionsApiBaseTests extends ControllerTestCommon with BeforeAndAfterEac
         WhiskActionMetaData(
           actionName.path,
           actionName.name,
-          js6MetaData(),
+          js6MetaData(binary = false),
```
Comment (dubeejw) [inline tests/src/test/scala/whisk/core/controller/test/WebActionsApiTests.scala]: The binary field could be optional and default to a value of `false`.

File: tests/src/test/scala/whisk/core/entity/test/ExecHelpers.scala [hunk 2]
```diff
@@ -80,6 +82,12 @@ trait ExecHelpers extends Matchers with WskActorSystem with StreamLogging {
     CodeExecAsAttachment(manifest, attachment, main.map(_.trim))
   }
 
+  protected def javaMetaData(main: Option[String] = None, binary: Boolean) = {
```
Comment (dubeejw) [inline tests/src/test/scala/whisk/core/entity/test/ExecHelpers.scala]: `binary` can be optional and default to `true`.

**Comment by dubeejw** [comment]
@rabbah, any comments on this one?

File: common/scala/src/main/scala/whisk/core/entity/Exec.scala [hunk 3]
```diff
@@ -95,9 +95,23 @@ sealed abstract class CodeExec[+T <% SizeConversion] extends Exec {
...
   /** Indicates if a container image is required from the registry to execute the action. */
   val pull: Boolean
 
+  /**
+   * Indicates whether the code is stored in a text-readable or binary format.
+   * The binary bit may be read from the database but currently it is always computed
+   * when the "code" is moved to an attachment this may get changed to avoid recomputing
+   * the binary property.
+   */
+  val binary: Boolean
```
Comment (rabbah) [inline common/scala/src/main/scala/whisk/core/entity/Exec.scala]: do you need the other properties?

File: tests/src/test/scala/whisk/core/controller/test/ActionsApiTests.scala [hunk 4]
```diff
@@ -159,26 +159,136 @@ class ActionsApiTests extends ControllerTestCommon with WhiskActionsApi {
...
-      Get(path) ~> Route.seal(routes(creds)) ~> check {
-        status should be(OK)
-        val response = responseAs[JsObject]
-        response.fields("exec").asJsObject.fields("code") should be("??".toJson)
-        responseAs[WhiskAction] shouldBe a[WhiskAction]
-      }
+        Put(s"$collectionPath/${action.name}", content) ~> Route.seal(routes(creds)) ~> check {
+          status should be(OK)
+          val response = responseAs[WhiskAction]
+          response should be(expectedWhiskAction)
```
Comment (rabbah) [inline tests/src/test/scala/whisk/core/controller/test/ActionsApiTests.scala]: we should perhaps change the PUT response to just be 200/201 in the future (for all the created entities).

Comment (dubeejw) [reply tests/src/test/scala/whisk/core/controller/test/ActionsApiTests.scala]: Opened an issue for this https://github.com/apache/incubator-openwhisk/issues/3274.

File: common/scala/src/main/scala/whisk/core/entity/Exec.scala (hunk 3 above)
Comment (dubeejw) [reply common/scala/src/main/scala/whisk/core/entity/Exec.scala]: Hmm, I think they all should be there except `code`.

**Pull request merged**